src/pages/index.js

```diff
@@ -137,18 +205,71 @@ function closeModal(modal) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  const submitBtnTwo = evt.submitter;
+  setButtonText(submitBtnTwo, true, "Saving...", "Save");
+  //submitBtnTwo.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      profileAvatar.src = data.avatar;
+      evt.target.reset();
+      disableButton(submitBtn, config.inactiveButtonClass);
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtnTwo, false, "Saving...", "Save");
+    });
 }
 ////////////////////////////////////
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
+  const submitButton = evt.submitter;
+
   const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  evt.target.reset();
-  closeModal(cardModal);
+  setButtonText(submitButton, true, "Save", "Saving...");
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      disableButton(submitButton, config.inactiveButtonClass);
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false, "Save", "Saving..."); // Reset button text after request
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const avatarSubmitButton = evt.submitter;
+
+  setButtonText(avatarSubmitButton, true, "Save", "Saving...");
+  api
+    .editAvatarInfo({ avatar: avatarInput.value })
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+
+      evt.target.reset();
+      closeModal(avatarModal);
+      console.log("Avatar submit button:", avatarSubmitButton);
+
+      // Disable button after successful operation
+      disableButton(avatarSubmitButton, "modal__submit-btn_disabled");
```

Comment:
`config.inactiveButtonClass` should be used here not to hardcode the validation string


```js

      disableButton(avatarSubmitButton, config.inactiveButtonClass);

```